suzieq/poller/worker/nodes/node.py

```diff
@@ -525,186 +415,183 @@ async def _parse_device_type_hostname(self, output, _) -> None:
+            elif self.devtype == "iosxr":
+                self.__class__ = IosXRNode
+            elif self.devtype == "iosxe":
+                self.__class__ = IosXENode
+            elif self.devtype == "ios":
+                self.__class__ = IOSNode
+            elif self.devtype.startswith("junos"):
+                self.__class__ = JunosNode
+            elif self.devtype == "nxos":
+                self.__class__ = NxosNode
+            elif self.devtype.startswith("sonic"):
+                self.__class__ = SonicNode
+            elif self.devtype == "panos":
+                self.__class__ = PanosNode
+            elif self.devtype == "linux":
+                self.__class__ = LinuxNode
 
         # Now invoke the class specific NOS version extraction
         if version_str:
             self._extract_nos_version(version_str)
 
-    def set_unreach_status(self):
-        '''Mark node communication as down'''
-        self._status = "unreachable"
-
-    def set_good_status(self):
-        '''Mark node as reachable'''
-        self._status = "good"
-
-    def is_alive(self):
-        '''Is node alive'''
-        return self._status == "good"
-
-    def set_hostname(self, hostname=None):
+    def _set_hostname(self, hostname=None):
         '''Set node hostname'''
         if hostname:
             self.hostname = hostname
 
-    async def init_boot_time(self):
-        """Fill in the boot time of the node by executing certain cmds"""
-        await self.exec_cmd(self._parse_boottime_hostname,
-                            ["cat /proc/uptime", "hostname",
-                             "cat /etc/os-release"], None, 'text')
+    async def _init_ssh(self, init_dev_data=True, rel_lock=True) -> None:
+        '''Setup a persistent SSH connection to our node.
 
-    def post_commands(self, service_callback, svc_defn: dict,
-                      cb_token: RsltToken):
-        '''Return the command output back to the service that requested it'''
-        if cb_token:
-            cb_token.nodeQsize = self._service_queue.qsize()
-        self._service_queue.put_nowait([service_callback, svc_defn, cb_token])
+        In some cases such as IOSXE/XR, this may not actually setup a
+        persistent SSH session, but does the basics of setting one up.
 
-    async def run(self):
-        '''Main workhorse routine for Node'''
+        rel_lock is a critical parameter to avoid deadlocks. If too
+        many services attempt to init_ssh at the same time, we'll cause
+        much heartache. So, we acquire a lock to ensure this is done only
+        once.
 
-        tasks = []
-        while True:
-            if self.sigend:
-                await self._terminate()
+        However, when the calling party has additional work to do, the
+        caller MUST acquire the lock and pass rel_lock=False. And the
+        caller MUST ensure they release the lock.
+        '''
+        if self._retry and not self._conn:
+            if rel_lock:
```

Comment:
In the previous version of the code `rel_lock` argument told the function not to release the lock, but in any case the lock was acquired. At this point I think still calling this function `rel_lock` is misleading, we might call it something like `acquire_lock`

Reply:
Changed to use_lock